src/gui/guiFormSpecMenu.cpp

```diff
@@ -4320,8 +4320,10 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 		case BET_DOWN: {
 			// Some mouse button has been pressed
 
+#ifndef HAVE_TOUCHSCREENGUI
 			if (m_held_mouse_button != BET_OTHER)
 				break;
+#endif
 			if (button == BET_LEFT || button == BET_RIGHT || button == BET_MIDDLE)
 				m_held_mouse_button = button;
```

Comment:
Doesn't this "corrupt" the internal state, e.g. by setting `m_held_mouse_button` to `BET_RIGHT` while we're already left-dragging?